mods/taskbar-empty-space-clicks.wh.cpp

```diff
@@ -1697,40 +2083,69 @@ BOOL Wh_ModInit()
                                 g_pDeviceEnumerator.put_void())) ||
         !g_pDeviceEnumerator)
     {
-        Wh_Log(L"Failed to create DeviceEnumerator COM instance, ModInit failed");
-        return FALSE;
+        // this is not mandatory, if failed the volume mute feature will not be available
+        LOG_ERROR(L"Failed to create DeviceEnumerator COM instance. Volume mute feature will not be available!");
     }
     else
     {
-        Wh_Log(L"DeviceEnumerator COM initilized");
+        LOG_INFO(L"DeviceEnumerator COM initilized");
     }
 
-    // optional feature, not critical for the mod
-    if (!FindDesktopWindow())
+    // sometimes directly after boot the desktop window is not found, so try to search for it few times
+    bool desktopWindowFound = false;
+    for (int i = 0; i < 10; i++)
     {
-        Wh_Log(L"Failed to find Desktop window. Hide icons feature will not be available!");
+        if (FindDesktopWindow())
+        {
+            desktopWindowFound = true;
+            break;
+        }
+        std::this_thread::sleep_for(std::chrono::milliseconds(50));
```

Comment:
`Wh_ModInit` blocks the process execution if the mod is loaded before the process (explorer.exe) started executing. In this case, the desktop window will never be found, and this will just delay the initialization. IMO it's best to find it lazily when the relevant action is performed.